cranelift-codegen/meta/src/shared/instructions.rs

```diff
@@ -394,6 +404,17 @@ pub fn define(
         .can_trap(true),
     );
 
+    ig.push(
+        Inst::new(
+            "trapnf",
+            r#"
+        A non-fatal trap.
```

Comment:
"trapnf" is pretty similar to "trapnz", such that it could be confused for a kind of conditional trap. The phrase "trap on non-fatal" comes to mind, though of course that doesn't make sense ;-). How about we rename this to "resumable_trap"?

Also, it'd be good to put some of your explanation of what a non-fatal trap is above in the documentation here.